### PR TITLE
improve ads of kmesh

### PR DESCRIPTION
**Pull request by YaoZengzeng**
1. refactor ads of kmesh, throw away kmesh.json
2. automatically generate node id
3. setup secure connection with istiod

**What type of PR is this?**

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes part of #75

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

**Comment by YaoZengzeng** [comment]
@hzxuzhonghu @nlgwcy @kwb0523 could review the PR first.

File: build/docker/clusterrolebinding.yaml [hunk 1]
```diff
@@ -11,3 +11,4 @@ roleRef:
 subjects:
 - kind: ServiceAccount
   name: kmesh
+  namespace: default
```
Comment (hzxuzhonghu) [inline build/docker/clusterrolebinding.yaml:14]: Good catch,   we should by default install in a `kmesh-system`

File: pkg/controller/envoy/ads_config.go [hunk 2]
```diff
@@ -59,179 +52,57 @@ var (
...
 	return &config
 }
 
 func (c *XdsConfig) SetClientArgs() error {
-	flag.StringVar(&c.File, "config-file",
-		"/etc/kmesh/kmesh.json", "[if -enable-kmesh] deploy in kube cluster")
-	flag.StringVar(&c.ServiceNode, "service-node", "TODO", "[if -enable-kmesh] TODO")
-	flag.StringVar(&c.ServiceCluster, "service-cluster", "TODO", "[if -enable-kmesh] TODO")
 	flag.BoolVar(&c.EnableAds, "enable-ads", true, "[if -enable-kmesh] enable control-plane from ads")
+	flag.StringVar(&c.ServiceCluster, "service-cluster", "TODO", "[if -enable-kmesh] TODO")
```
Comment (hzxuzhonghu) [inline pkg/controller/envoy/ads_config.go:67]: remove. we donot need to specify it

File: pkg/controller/envoy/ads_config.go [hunk 3]
```diff
@@ -59,179 +52,57 @@ var (
...
 }
 
 func GetConfig() *XdsConfig {
 	return &config
 }
 
 func (c *XdsConfig) SetClientArgs() error {
-	flag.StringVar(&c.File, "config-file",
-		"/etc/kmesh/kmesh.json", "[if -enable-kmesh] deploy in kube cluster")
-	flag.StringVar(&c.ServiceNode, "service-node", "TODO", "[if -enable-kmesh] TODO")
```
Comment (hzxuzhonghu) [inline pkg/controller/envoy/ads_config.go:76]: update kmesh_commands document as well

File: pkg/controller/envoy/ads_config.go [hunk 4]
```diff
@@ -59,179 +52,57 @@ var (
...
 		return nil
 	}
 
-	if c.File, err = filepath.Abs(c.File); err != nil {
-		return err
-	}
+	podIP := env.Register("INSTANCE_IP", "", "").Get()
+	podName := env.Register("POD_NAME", "", "").Get()
+	podNamespace := env.Register("POD_NAMESPACE", "", "").Get()
+	discoveryAddress := env.Register("MESH_CONTROLLER", "istiod.istio-system.svc:15012", "").Get()
```
Comment (hzxuzhonghu) [inline pkg/controller/envoy/ads_config.go:79]: MESH_CONTROLLER is removed from deployment

**Review by hzxuzhonghu — APPROVED** [review]
is there a follow up to support delta xds

**Comment by YaoZengzeng** [comment]
ping  @nlgwcy @kwb0523

**Review by hzxuzhonghu — COMMENTED** [review]
/lgtm

**Pull request merged**